bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -422,11 +427,14 @@ private void updateMeter(ShellySettingsStatus status, int id, ShellySettingsMete
         relayStatus.meters.set(id, sm);
     }
 
-    private boolean updateEmStatus(ShellySettingsStatus status, @Nullable Shelly2StatusEm1 em, boolean channelUpdate)
-            throws ShellyApiException {
+    private boolean updateEmStatus(int id, ShellySettingsStatus status, @Nullable Shelly2StatusEm1 em,
+            boolean channelUpdate) throws ShellyApiException {
         if (em == null) {
             return false;
         }
+        if (em.id == null) { // invalid in fw 1.6.1
+            em.id = id >= 10 ? id - 10 : id; // ids start at 10
```

Comment:
This looks complicated. The method is called with 0, 10 and 11, which is then "translated" to 0, 0 and 1, respectively. Why? Are em0 and em10 to be handled in the same way? Perhaps simply call with "final" id's rather than having this "translation"?